### ISSUE TITLE
Argobots 1.1 Release

### ISSUE THREAD

**Issue by shintaro-iwasaki** [pmodels/argobots#304]
Argobots 1.1 will be released around the end of March (the release date will be March 31).
The following will be expected on top of Argobots 1.1b1.
 -  ~Fix #93~ [Fixed by #311]
 - ~Better interoperability interface (combining multiple Argobots-aware components)~
   - ~Fix #86~ [Not in Argobots 1.1]
   - ~(Maybe we need a virtual execution stream)~ [We do not plan to add this feature to Argobots 1.1]
 - ~API/ABI compatibility check with Argobots 1.0~ [All tests passed: #307]
   - ~Adding a new test~ [CI is added]
 - ~Memory leak check~
   - ~Adding a new test~ [Fixed by #309]
   - ~Coverity check~ [CI is added. Outstanding issues have been fixed: #308]
   - ~Fix some functions such as `ABT_init()` and `ABT_xstream_create()` that do not free memory/resources on error (#116).~ [Fixed by #309]
 - ~Specification improvement~ [This does not directly affect the Argobots runtime. We will update them during the normal Argobots development].
   - Proofread the API specification
   - Environmental variables
 
The following will *not* be included in Argobots 1.1
 - #154
 - #155

Any feedback is welcome!

**Comment by shintaro-iwasaki** [comment]
I set up CI for ABI/API compatibility check. This will be checked by the ABI Compliance Checker (https://lvc.github.io/abi-compliance-checker/).
It can be triggered manually if necessary. For example, changes in #307 break the compatibility, so the test fails.
https://jenkins-pmrs.cels.anl.gov/job/argobots-review-abicompat/2/
The report looks like this:
https://jenkins-pmrs.cels.anl.gov/job/argobots-review-abicompat/2/compiler=gnu4.8.5,context=fcontext,jenkins_configure=default,label=argobots_centos64,release_ver=1.0/artifact/compat_report.html

It will also be tested nightly. Now the current Argobots main passes all the tests (https://jenkins-pmrs.cels.anl.gov/view/abt/job/argobots-main-abicompat-nightly/).

**Comment by shintaro-iwasaki** [comment]
Coverity check was added. Argobots will be checked regularly by CI under the following configurations.
```sh
# Default configuration
./configure
# For performance optimization
./configure --enable-perf-opt
# Configuration for debugging
./configure --enable-debug=yes --enable-stack-overflow-check=canary-16
# When a machine does not support fcontext (e.g., non-POWER/ARM64/AMD64 machine)
./configure --disable-fcontext
# Experimental 2.0 API
./configure --enable-ver20-api
# Uncommon setting: --enable-feature=no
./configure --enable-feature=no
# Other minor settings
./configure --enable-fcontext --enable-dynamic-promotion --enable-affinity \
            --enable-wait-policy=active --enable-tool
```
#308 fixed bugs found by Coverity. Now there's no outstanding issue.

**Comment by shintaro-iwasaki** [comment]
Fixing #86 requires significant changes around the primary execution stream. We decided not to include this in Argobots 1.1.

**Comment by shintaro-iwasaki** [comment]
#316 is a major issue that needs to be fixed, so I would like to have another PR before Argobots 1.1.
We will release Argobots 1.1rc2 after fixing this issue. Possibly #314 is added too.

**Comment by shintaro-iwasaki** [comment]
#316 has been fixed by #317.
APIs requested by #314 were added by #319.
